CenterForOpenScience/ember-osf-web
# [EMB-379] Resolve guids in mirage

## Purpose
Allow resolving guids in mirage, so we can write acceptance tests for pages that depend on that (e.g. file detail page)


## Summary of Changes
- Generate pseudo-random guids in mirage for users, files, nodes, and registrations
  - Same sequence of guids for each type, so if an object is created in the default scenario, the guid will stay valid across reloads
- Add shorthands for configuring route handlers for our API, inspired by mirage's shorthands
  - `osfResource` -- similar to [`this.resource`](http://www.ember-cli-mirage.com/docs/v0.4.x/shorthands/#resource-helper), adds handlers for a list of objects, detail view, create, update, delete
  - `osfNestedResource` -- for adding nested routes that correspond to a relationship
- Add `ApplicationSerializer.buildRelationships()` as a more flexible way of serializing relationships than relying on mirage's unintuitive use of `links()`
  - Model serializers should override to build their relationships hash
  - Required for the weird format of guids' `referent` relationship


## Side Effects
n/a


## Feature Flags
n/a


## QA Notes
No testing by QA, but the reviewer could run it locally and check that browsing around with mirage works. Especially check going directly to the file detail page (http://localhost:4200/mxxdb/ ) to see the guid resolve.


## Ticket


https://openscience.atlassian.net/browse/EMB-379

# Reviewer Checklist
